src/uwtools/drivers/forecast.py

```diff
@@ -256,4 +316,106 @@ def _mpi_env_variables(self, delimiter: str = " ") -> str:
+    """
+
+    def __init__(
+        self,
+        config_file: str,
+        dry_run: bool = False,
+        batch_script: Optional[str] = None,
+    ):
+        """
+        Initialize the Forecast Driver.
+        """
+
+        super().__init__(config_file=config_file, dry_run=dry_run, batch_script=batch_script)
+
+    def create_namelist(self, cycle: datetime, output_path: OptionalPath) -> None:
+        """
+        Uses an object with user supplied values and an optional namelist base file to create an
+        output namelist file. Will "dereference" the base file.
+
+        :param cycle: The date and start time for the forecast.
+        :param output_path: Optional location of output namelist.
+        """
+        self._create_user_updated_config(
+            config_class=NMLConfig,
+            config_values=self._config["namelist"],
+            output_path=output_path,
+        )
+        start_time = cycle.strftime("%Y-%m-%d_%H:%M:%S")
+        date_values = {
+            "nhyd_model": {
+                "config_start_time": start_time,
+            },
+        }
+        log.info(f"Updating namelist date values to start at: {start_time}")
+        config_obj = NMLConfig(output_path)
+        config_obj.update_values(date_values)
+        config_obj.dump(output_path)
+
+    def create_streams(self, output_path: OptionalPath) -> None:
+        """
+        Create the streams file from a template.
+
+        :param output_path: Optional location of output streams file.
+        """
+
+        template_file = self._config["streams"]["template"]
+        values = self._config["streams"]["vars"]
+
+        with open(template_file, "r", encoding="utf-8") as f:
+            template_str = f.read()
+
+        template = J2Template(values=values, template_str=template_str)
+        with writable(output_path) as f:
+            print(template.render(), file=f)
+
+    @property
+    def schema_file(self) -> Path:
+        """
+        The path to the file containing the schema to validate the config file against.
+
+        :return: The string path to the schema file.
+        """
+        with resources.as_file(resources.files("uwtools.resources")) as path:
+            return path / "MPASForecast.jsonschema"
+
+    # Private methods
+
+    def _define_boundary_files(self) -> Dict[str, str]:
+        """
+        No boundary files are currently needed for MPAS global support.
+
+        :return: A dict of boundary file names mapped to source input file paths
+        """
+        return {}
+
+    def _mpi_env_variables(self, delimiter=" ") -> str:
+        """
+        Sets the environment variables needed for running the mpi command.
+
+        :param delimiter: The delimiter to be used between items in the configuration dictionary.
+        :return: A bash-formatted string of MPI environment variables.
+        """
+        return delimiter.join([f"{k}={v}" for k, v in self._config.get("mpi_settings", {}).items()])
```

Comment:
Just trying to get my bearings: Is `mpi_settings` something that should appear in `MPASForecast.jsonschema`? I don't see the `additionalProperties` key in that schema, so I suppose additional, unrecognized content might be tolerated, but that's probably not what we'd want. I was wondering, again, if this item was really optional (i.e. why we're using `.get()` with a `{}` default instead of `[]`), and this time thought that -- maybe unlike, say, a block defining the content of `model_configure` for FV3 -- this might truly be optional, so thought I'd see if the schema shed any light on this. But it did not.

Reply:
I think probably I should add it there, but really it would just be a generic object to hold arbitrary environment variables.